### ISSUE TITLE
Create a todocheck-bot for reopening issues on github if unclosed TODOs are present

### ISSUE THREAD

**Issue by preslavmihaylov** [preslavmihaylov/todocheck#57]
The most tedious part of using `todocheck` is that when someone closes an issue as they assume it's done, all future todocheck CI steps start to fail as there are TODOs referenced to a closed issue.

This requires someone to manually reopen the issue in order to unblock all failing CI pipelines.

By having a bot automate this process, this would improve user experience \w `todocheck` tremendously.

The scope of this task is to create a github action called todocheck-bot, which triggers on "issue closed" events, runs todocheck against the repository & reopens the issue if there are TODOs linked against the closed issue.

This is a big undertaking & might even require a separate repository and/or several follow-up tasks. But this project seems to be possible using Github actions, similar to [this project](https://github.com/imjohnbo/issue-bot)

**Comment by bengsparks** [comment]
I would be interested in taking part in this issue. Has any work on this started yet?

**Comment by preslavmihaylov** [comment]
Ahh, you're right. I've opened a follow-up issue for that - #162

Feel free to take that up as well if interested.

**Comment by bengsparks** [comment]
The first working proof of concept can be found here:
https://github.com/bengsparks/todocheck-bot/releases/tag/v0.0.2.

It currently does nothing else except just reopening prematurely closed issues and lacks QoL features such as commenting why the issue was suddenly reopened.
Another interesting avenue to pursue might be to prevent merging PRs if issues that would be closed thereby are still in the codebase.

But I suppose now that GitHub is at least supported we can move discussion to repository mentioned above?
